core/environment/environment.go

```diff
@@ -614,8 +614,8 @@ func newEnvironment(userVars map[string]string, newId uid.ID) (env *Environment,
 					errorMsg = e.Err.Error()
 				}
 
-				if e.Event == "STOP_ACTIVITY" {
-					// If the event is STOP_ACTIVITY, we remove the active run number after all hooks are done.
+				if e.Event == "STOP_ACTIVITY" || e.Event == "GO_ERROR" {
+					// If the event is STOP_ACTIVITY or GO_ERROR, we remove the active run number after all hooks are done.
 					env.workflow.GetVars().Set("last_run_number", strconv.Itoa(int(env.currentRunNumber)))
```

Comment:
i think we should avoid setting `"last_run_number"` if `env.currentRunNumber` is 0. This could happen if we e.g. reach ERROR without ever starting a run.

This way, it will be less risky to have a bug due to some piece of code assuming that non-empty `last_run_number` means we had a run before.